### PR TITLE
Fix: Failed strikes should always fail movement

### PR DESCRIPTION
**Pull request by eddie-knight**
- Added condition for failed movements to ExecuteMovement
- Added unit test for ExecuteMovement

tertiary: fixed malfunctioning CI pipeline so that checks will actually run

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     paths-ignore:
       - "**/*.md"
 
-permissions: read
+permissions: read-only
```
Comment (grudra7714) [inline .github/workflows/ci.yml:8]: Maybe i am looking at the wrong spot but i dont see a `read-only` permission https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

**Review by grudra7714 — COMMENTED** [review]
See comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (eddie-knight) [reply .github/workflows/ci.yml:8]: bleerrrrrg. you right. gonna revert this and see what the previous error was.

Comment (eddie-knight) [reply .github/workflows/ci.yml:8]:
```suggestion
permissions: read
```

Comment (eddie-knight) [reply .github/workflows/ci.yml]: fixed

**Review by grudra7714 — APPROVED** [review]

**Pull request merged**